parcels/_datasets/utils.py

```diff
@@ -67,32 +67,50 @@ def dataset_repr_diff(ds1: xr.Dataset, ds2: xr.Dataset) -> str:
     return "".join(diff)
 
 
-def compare_datasets(ds1, ds2, ds1_name="Dataset 1", ds2_name="Dataset 2"):
+def _dicts_equal(d1, d2):
+    # compare two dictionaries, including when their entries are lists or arrays ( == throws an error then)
+    if d1.keys() != d2.keys():
+        return False
+    for k in d1:
+        v1, v2 = d1[k], d2[k]
+        # Compare lists or arrays element-wise
+        if isinstance(v1, (list, np.ndarray)) and isinstance(v2, (list, np.ndarray)):
+            if not np.array_equal(np.array(v1), np.array(v2)):
+                return False
+        else:
+            if v1 != v2:
+                return False
+    return True
+
+
+def compare_datasets(ds1, ds2, ds1_name="Dataset 1", ds2_name="Dataset 2", verbose=True):
     print(f"Comparing {ds1_name} and {ds2_name}\n")
 
-    # Compare dataset attributes
-    print("Dataset Attributes Comparison:")
-    if ds1.attrs == ds2.attrs:
-        print("  Dataset attributes are identical.")
-    else:
-        print("  Dataset attributes differ.")
-        for attr_name in set(ds1.attrs.keys()) | set(ds2.attrs.keys()):
-            if attr_name not in ds1.attrs:
-                print(f"    Attribute '{attr_name}' only in {ds2_name}")
-            elif attr_name not in ds2.attrs:
-                print(f"    Attribute '{attr_name}' only in {ds1_name}")
-            elif ds1.attrs[attr_name] != ds2.attrs[attr_name]:
-                print(f"    Attribute '{attr_name}' differs:")
-                print(f"      {ds1_name}: {ds1.attrs[attr_name]}")
-                print(f"      {ds2_name}: {ds2.attrs[attr_name]}")
-    print("-" * 30)
+    if verbose:
```

Comment:
Did you only intend to hide the `"  Dataset attributes are identical."` behind the verbose flag? This would also hide the dataset level attribute comparison.

To reduce the branching we could do something like the following? Would that look better in the main function body so that individual statements can just be flagged as verbose?
```python
def compare_datasets(ds1, ds2, ds1_name="Dataset 1", ds2_name="Dataset 2", verbose=True):
    print(f"Comparing {ds1_name} and {ds2_name}\n")
    def verbose_print(*args, **kwargs):
        if verbose:
            print(*args, **kwargs)
    
    # use verbose_print when printing things we only want in verbose output
```

Reply:
> I hadn't included the dataset-level attributes in the circulation models, which is why I commented this out. But your suggestion is indeed nicer, so I'll change the code

Gotcha - sounds good